src/sentry/static/sentry/app/views/organizationIntegrationConfig.jsx

```diff
@@ -0,0 +1,259 @@
+    this.dialog = null;
+    window.addEventListener('message', this.receiveMessage, false);
+  }
+
+  componentWillUnmount() {
+    super.componentWillUnmount();
+    window.removeEventListener('message', this.receiveMessage);
+
+    if (this.dialog !== null) {
+      this.dialog.close();
+    }
+  }
+
+  getTitle() {
+    if (this.state.config === null) {
+      return 'Global Integrations';
+    }
+
+    const {providerKey} = this.props.params;
+    const provider = this.state.config.providers.find(p => p.key == providerKey);
+
+    return `${provider.name} Integration`;
+  }
+
+  getEndpoints() {
+    const {orgId} = this.props.params;
+
+    return [
+      ['config', `/organizations/${orgId}/config/integrations/`],
+      ['itemList', `/organizations/${orgId}/integrations/`, {query: {status: ''}}],
+    ];
+  }
+
+  launchAddIntegration = provider => {
+    const name = 'sentryAddIntegration';
+
+    const {url, width, height} = provider.setupDialog;
+    const {left, top} = computeCenteredWindow(width, height);
+
+    this.dialog = window.open(
+      url,
+      name,
+      `scrollbars=yes,width=${width},height=${height},top=${top},left=${left}`
+    );
+
+    this.dialog.focus();
+    this.dialog.onclose = () => document.location.refresh();
+  };
+
+  receiveMessage = message => {
+    if (message.origin !== document.origin) {
+      return;
+    }
+
+    if (message.source !== this.dialog) {
+      return;
+    }
+
+    this.dialog = null;
+
+    const {success, data} = message.data;
+
+    if (!success) {
+      // TODO: Error state here
+      return;
+    }
+
+    let itemList = [...this.state.itemList, data];
+    itemList = sortArray(itemList, i => i.name);
+    itemList = _.uniqBy(itemList, i => i.id);
+
+    this.setState({itemList});
+  };
+
+  deleteIntegration = integration => {
+    const {orgId} = this.props.params;
+
+    const options = {
+      method: 'DELETE',
+      success: () =>
+        this.setState({
+          itemList: this.state.itemList.filter(item => item.id !== integration.id),
+        }),
+
+      // TODO toast indoactor for errors / complete
+    };
+
+    this.api.request(`/organizations/${orgId}/integrations/${integration.id}/`, options);
+  };
+
+  renderBody() {
+    const {orgId, projectId, providerKey} = this.props.params;
+
+    const integrations = this.state.itemList.filter(i => i.provider.key === providerKey);
+    const provider = this.state.config.providers.find(p => p.key == providerKey);
+
+    // TODO: some kind of 404ish thing here
+
+    // TODO: Workspaces needs to get genericised into the integration
+
+    const titleIcon = <PluginIcon size={28} pluginId={provider.key} />;
+
+    const header = (
+      <PanelHeader disablePadding hasButtons>
+        <Flex align="center">
+          <Box px={2} flex="1">
+            {t('Workspaces')}
+          </Box>
+          <Box mr={1}>
+            <Button size="xsmall" onClick={() => this.launchAddIntegration(provider)}>
```

Comment:
Let's use `handle*` prefix for callback handlers

Reply:
8e6cfe6